service/lxd.go

```diff
@@ -333,10 +333,10 @@ func (s *LXDService) GetResources(ctx context.Context, target string, address st
 	return client.GetServerResources()
 }
 
-// GetUplinkInterfaces returns a map of peer name to slice of api.Network that may be used with OVN.
-func (s LXDService) GetUplinkInterfaces(ctx context.Context, bootstrap bool, peers []mdns.ServerInfo) (map[string][]api.Network, error) {
-	clients := map[string]lxd.InstanceServer{}
-	networks := map[string][]api.Network{}
+// getNetworkInterfaces fetches network candidates based on peer information and filtering rules contained in the collector function.
+func getNetworkInterfaces[T api.Network | CephDedicatedInterface](s LXDService, ctx context.Context, bootstrap bool, peers []mdns.ServerInfo, collectFn func(c map[string]lxd.InstanceServer, acc map[string][]T, peer string, nets []api.Network)) (map[string][]T, error) {
```

Comment:
Do we need to have this much complexity here? I.e. would it be simpler to just have a different function when we want to return a different type?

Reply:
yes, I can try to simplify!

Reply:
fixed

Reply:
Did you push your changes? This still looks the same to me.